src/frontend/src/planner/relation.rs

```diff
@@ -93,10 +97,92 @@ impl Planner {
+                let db_name = session.database();
+                let catalog_reader = session.env().catalog_reader().read_guard();
+                let mut source_catalog = None;
+                for schema in catalog_reader.iter_schemas(db_name).unwrap() {
+                    if schema
+                        .get_table_by_id(&base_table.table_catalog.id)
+                        .is_some()
+                    {
+                        source_catalog = schema.get_source_by_name(
+                            &base_table.table_catalog.iceberg_source_name().unwrap(),
+                        );
+                        break;
+                    }
+                }
+                if let Some(source_catalog) = source_catalog {
+                    let column_map: HashMap<String, (usize, ColumnCatalog)> = source_catalog
+                        .columns
+                        .clone()
+                        .into_iter()
+                        .enumerate()
+                        .map(|(i, column)| (column.name().to_string(), (i, column)))
+                        .collect();
+                    let exprs = scan
+                        .table_catalog()
+                        .column_schema()
+                        .fields()
+                        .iter()
+                        .map(|field| {
+                            let source_filed_name = if field.name == ROWID_PREFIX {
+                                RISINGWAVE_ICEBERG_ROW_ID
+                            } else {
+                                &field.name
+                            };
+                            if let Some((i, source_column)) = column_map.get(source_filed_name) {
+                                if source_column.column_desc.data_type == field.data_type {
+                                    ExprImpl::InputRef(
+                                        InputRef::new(*i, field.data_type.clone()).into(),
+                                    )
+                                } else {
+                                    let mut input_ref = ExprImpl::InputRef(
+                                        InputRef::new(
+                                            *i,
+                                            source_column.column_desc.data_type.clone(),
+                                        )
+                                        .into(),
+                                    );
+                                    FunctionCall::cast_mut(
+                                        &mut input_ref,
+                                        field.data_type().clone(),
+                                        CastContext::Explicit,
+                                    )
+                                    .unwrap();
+                                    input_ref
+                                }
+                            } else {
+                                // fields like `_rw_timestamp`, would not be found in source.
```

Comment:
Is there any other example? 

I am think whether we should `bail!` if a column is not found in the Iceberg source and it's not `_rw_timestamp`

Reply:
I think `_rw_timestamp` is the only one that would hit this branch. 

> I am think whether we should bail! if a column is not found in the Iceberg source and it's not _rw_timestamp

This is the planner's beginning, so we must select all columns from a table (including the `_rw_timestamp`) unless we move this conversion to the end of the optimizer (after column pruning).